core/src/test/java/io/github/tcdl/msb/api/ChannelMonitorIT.java

```diff
@@ -37,10 +58,158 @@ public void testAnnouncement() throws InterruptedException {
             announcementReceived.countDown();
         };
 
-        DefaultChannelMonitorAggregator channelMonitorAggregator = new DefaultChannelMonitorAggregator(msbContext, new ScheduledThreadPoolExecutor(1), handler);
+        channelMonitorAggregator = msbContext.getObjectFactory().createChannelMonitorAggregator(handler);
         channelMonitorAggregator.start(false, ChannelMonitorAggregator.DEFAULT_HEARTBEAT_INTERVAL_MS, ChannelMonitorAggregator.DEFAULT_HEARTBEAT_TIMEOUT_MS);
 
         channelMonitorAgent.producerTopicCreated(TOPIC_NAME);
         assertTrue("Announcement was not received", announcementReceived.await(RequesterResponderIT.MESSAGE_TRANSMISSION_TIME, TimeUnit.MILLISECONDS));
     }
+
+    @Test
+    public void testAnnouncementUnexpectedMessage() throws InterruptedException {
+        String TOPIC_NAME = "topic1";
+        ChannelMonitorAgent channelMonitorAgent = new DefaultChannelMonitorAgent(msbContext);
+
+        CountDownLatch announcementReceived = new CountDownLatch(1);
+        Callback<AggregatorStats> handler = stats -> {
+            assertTrue(stats.getTopicInfoMap().containsKey(TOPIC_NAME));
+            assertEquals(1, stats.getTopicInfoMap().get(TOPIC_NAME).getProducers().size());
+            announcementReceived.countDown();
+        };
+
+        //simulate unexpectedMessage was consumed from in TOPIC_ANNOUNCE namespace
+        MockAdapter.pushRequestMessage(Utils.TOPIC_ANNOUNCE,
+                Utils.toJson(TestUtils.createMsbRequestMessageWithSimplePayload(Utils.TOPIC_ANNOUNCE), msbContext.getPayloadMapper()));
+
+        channelMonitorAggregator = msbContext.getObjectFactory().createChannelMonitorAggregator(handler);
+        channelMonitorAggregator.start(false, ChannelMonitorAggregator.DEFAULT_HEARTBEAT_INTERVAL_MS, ChannelMonitorAggregator.DEFAULT_HEARTBEAT_TIMEOUT_MS);
+        channelMonitorAgent.producerTopicCreated(TOPIC_NAME);
+
+        assertTrue("Announcement was not received", announcementReceived.await(RequesterResponderIT.MESSAGE_TRANSMISSION_TIME * 2, TimeUnit.MILLISECONDS));
```

Comment:
Is it a valid test? You're sending 2 messages and waiting for only 1 of them to be consumed. But await is unblocked until ANY of the responses arrive. 

So let's consider a case where our code is broken so that the handler is invoked even for an invalid message. In this scenario the verificaion will still pass without waiting for the second message. Am I missing something?